Dockerfile

```diff
@@ -1,13 +1,17 @@
-FROM ubuntu:16.04
-MAINTAINER Changxu Wang <wang_changxu@zju.edu.cn>
+#Copyright(c) Microsoft Corporation.All rights reserved.
+#Licensed under the MIT license.
 
-RUN apt-get update -y
-RUN apt-get install -y g++ cmake libboost-dev libgoogle-perftools-dev
+FROM ubuntu:jammy
 
-COPY . /opt/nsg
+RUN apt update
+RUN apt install -y software-properties-common
+RUN add-apt-repository -y ppa:git-core/ppa
+RUN apt update
+RUN DEBIAN_FRONTEND=noninteractive apt install -y git make cmake g++ libaio-dev libgoogle-perftools-dev libunwind-dev clang-format libboost-dev libboost-program-options-dev libmkl-full-dev libcpprest-dev python3.10
 
-WORKDIR /opt/nsg
-
-RUN mkdir -p build && cd build && \
-    cmake -DCMAKE_BUILD_TYPE=Release .. && \
-    make -j $(nproc)
+WORKDIR /app
+RUN git clone https://github.com/microsoft/DiskANN.git 
+WORKDIR /app/DiskANN
+RUN mkdir build
+RUN cmake -S . -B build  -DCMAKE_BUILD_TYPE=Release
+RUN cmake --build build
```

Comment:
I just found out this actually should be:

`RUN cmake --build build -- -j` for parallel builds.  I wasted so much time until I found this out.

Reply:
added